db/error_handler.cc

```diff
@@ -466,10 +466,13 @@ void ErrorHandler::SetBGError(const Status& bg_status,
     Status bg_err(new_bg_io_err, severity);
     CheckAndSetRecoveryAndBGError(bg_err);
     recover_context_ = context;
+    bool auto_recovery = true;
```

Comment:
Should this taken into account the value of max_bgerror_resume_count since autorecovery may not be enabled?

Reply:
Good catch - fixed.